yivi-docs/blog/2025-07-01-post-mortem/index.md

```diff
@@ -0,0 +1,75 @@
+Also our Yivi frontend packages code was not using the `irma.app` domain for Android devices, but instead used an intent link that does not require the associated domain file to be hosted without redirects.
+
+[Yivi frontend packages code](https://github.com/privacybydesign/yivi-frontend-packages/blob/da76c44b0698563bc95c66c3552c2cb914747ec0/plugins/yivi-client/state-client.js#L296)
+```js
+  _getMobileUrl(sessionPtr) {
+    const json = JSON.stringify(sessionPtr);
+    switch (this._userAgent) {
+      case 'Android': {
+        // Universal links are not stable in Android webviews and custom tabs, so always use intent links.
+        const intent = `Intent;package=org.irmacard.cardemu;scheme=irma;l.timestamp=${Date.now()}`;
+        return `intent://qr/json/${encodeURIComponent(json)}#${intent};end`;
+      }
+      case 'iOS': {
+        return `https://irma.app/-/session#${encodeURIComponent(json)}`;
+      }
+      default: {
+        throw new Error('Device type is not supported.');
+      }
+    }
+  }
+```
+
+To mitigate the issue, we changed the DNS settings of `irma.app` back to the old server, allowing the Yivi app to open on iOS devices again. This was a temporary solution to ensure that users could continue to use the app while we worked on a more permanent fix.
+
+The `open.yivi.app` domain was not affected by this issue, as it was not part of the domain migration. The `open.yivi.app` domain is an alternative Universal Link that can be used to open the Yivi app on iOS devices. We provided a workaround for affected users by creating a page on `irma.app/-/session` that converts the `irma.app` Universal Links to `open.yivi.app` Universal Links, allowing them to continue using the app without any issues.
+
+This workaround will stay into place until we have fully migrated the `irma.app` domain to the new server and ensured that all Universal Links are working correctly. Apple states that it can take up to a week for devices to recheck the associated domain file, so we will monitor the situation closely and make any necessary adjustments as needed.
+
+Affected users can decide to reinstall the Yivi app from the App Store, which will ensure that the app is able to open Universal Links correctly. However, this is not necessary for existing users who already have the app installed, as they will still be able to open the app without any issues.
+
+## Next steps
+1. **Reevaluate domain migration strategy**: We will review our domain migration strategy to ensure that future migrations do not cause similar issues with Universal Links or other features.
+2. **Standardize Universal Links**: We will standardize the use of Universal Links across all platforms to ensure a consistent user experience and to prevent issues with redirects in the future.
```

Comment:
In the past, there was a reason to use an intent URI instead of an App/Universal Link on Android, as is documented in the comment in the code above. Not sure if that's still valid. The Intent URI has slightly lower security as it only checks the package name and not the app identifier (iOS) or hash (Android).

I don't think standardizing on universal links will do anything to "prevent issues with redirects in the future" (if anything it's the opposite).